eslint.config.mjs

```diff
@@ -0,0 +1,109 @@
+import { defineConfig } from "eslint/config";
+import globals from "globals";
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import tsParser from "@typescript-eslint/parser";
+
+export default defineConfig([{
+    ignores: [
+        "node_modules/*",
+        "build/*",
+        "docs/site/*",
+        "lib/*",
+        "eslint.config.mjs",
+        "build.mjs",
+    ]
+},{
+    languageOptions: {
+        globals: {
+            ...globals.browser,
+            ...globals.node,
+        },
+
+        ecmaVersion: 2018,
```

Comment:
2024 is the latest.  Any reason/preference for 2018?

Reply:
General paranoia about supporting not just the latest iteration of browsers.

What would be the benefit to using the latest version?

Reply:
2018 makes sense. I was raising it since if one were to change the ECMA compilation mode, this would be a good moment. 

After looking at the features I would say that 2022 might be a nice mix of useful features (e.g. class fields) that are also widely supported. https://compat-table.github.io/compat-table/es2016plus/ 

(but this was more of a question than a suggestion.)